File: app/src/main/java/com/berkatfaatulohalawa1711010164/facevoting/API/NetworkConnectionInterceptor.java
package com.berkatfaatulohalawa1711010164.facevoting.API;

import android.content.Context;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;

import androidx.annotation.NonNull;

import java.io.IOException;

import okhttp3.Interceptor;
import okhttp3.Request;
import okhttp3.Response;

class NetworkConnectionInterceptor implements Interceptor {
    private Context mContext;

    public NetworkConnectionInterceptor(Context context) {
        mContext = context;
    }

    @NonNull
    @Override
    public Response intercept(@NonNull Interceptor.Chain chain) throws IOException {
        if (!isConnected()) {
            throw new NoConnectivityException();
            // ini untuk custom exception 'NoConnectivityException'
        }
        Request.Builder builder = chain.request().newBuilder();
        return chain.proceed(builder.build());
    }
    @SuppressWarnings("DEPRECATION")
    public boolean isConnected(){
        ConnectivityManager connectivityManager = (ConnectivityManager) mContext.getSystemService(Context.CONNECTIVITY_SERVICE);
        NetworkInfo netInfo = connectivityManager.getActiveNetworkInfo();
        return (netInfo != null && netInfo.isConnected());
    }
}

File: android/facevoting/app/src/main/java/com/berkatfaatulohalawa1711010164/facevoting/API/NetworkConnectionInterceptor.kt
package com.berkatfaatulohalawa1711010164.facevoting.API

import android.content.Context
import android.net.ConnectivityManager
import okhttp3.Interceptor
import okhttp3.Response

internal class NetworkConnectionInterceptor(private val mContext: Context?) : Interceptor {

    override fun intercept(chain: Interceptor.Chain): Response {
        if (!isConnected()) {
            throw NoConnectivityException()
        }
        return chain.proceed(chain.request().newBuilder().build())
    }

    @Suppress("DEPRECATION")
    fun isConnected(): Boolean {
        val connectivityManager = mContext?.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
        val netInfo = connectivityManager.activeNetworkInfo
        return netInfo != null && netInfo.isConnected
    }
}
